test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/SkipSection.java

```diff
@@ -98,33 +99,30 @@ public static SkipSection parse(XContentParser parser) throws IOException {
 
     public static final SkipSection EMPTY = new SkipSection();
 
-    private final Version lowerVersion;
-    private final Version upperVersion;
+    private final List<VersionRange> versionRanges;
     private final List<String> features;
     private final String reason;
 
     private SkipSection() {
-        this.lowerVersion = null;
-        this.upperVersion = null;
+        this.versionRanges = new ArrayList<>();
         this.features = new ArrayList<>();
         this.reason = null;
     }
 
     public SkipSection(String versionRange, List<String> features, String reason) {
         assert features != null;
-        Version[] versions = parseVersionRange(versionRange);
-        this.lowerVersion = versions[0];
-        this.upperVersion = versions[1];
+        this.versionRanges = parseVersionRanges(versionRange);
+        assert versionRanges.size() >= 1;
```

Comment:
Would this be clearer?

```suggestion
        assert versionRanges.isEmpty() == false;
```